### PR TITLE
Add chainerx test in observation_aggregator

### PR DESCRIPTION
**Pull request by keisukefukuda**
This PR depends on #8508

This PR is a replacement of #8322 by @belldandyxtq 

This PR adds ChainerX tests into the observation_aggregator
This PR partially solves #8031. This PR depends on #8321

~NOTE: This PR is completed w.r.t. the code, but it causes SEGV depending on the order of test cases in test_observatgion_aggregator.py. I guess it is because of the behavior of ChainerX's device setting. We need a deeper investigation on this issue.~

fixed!

### CODE REVIEW

File: tests/chainermn_tests/extensions_tests/test_observation_aggregator.py [hunk 1]
```diff
@@ -26,64 +27,101 @@ def forward(self, x):
...
+                                            use_chainer_variable,
+                                            communicate_interval,
+                                            use_gpu=True)
+        else:
+            with pytest.raises(ValueError):
+                run_test_observation_aggregator(communicator, xp,
+                                                use_chainer_variable,
+                                                communicate_interval,
+                                                use_gpu=True)
+    del communicator
```
Comment (belldandyxtq) [inline tests/chainermn_tests/extensions_tests/test_observation_aggregator.py]: Unnecessary

Comment (keisukefukuda) [reply tests/chainermn_tests/extensions_tests/test_observation_aggregator.py]: fixed.

**Comment by shu65** [comment]
Jenkins, test this please!

**Comment by keisukefukuda** [comment]
Jenkins, test this please!

File: tests/chainermn_tests/extensions_tests/test_observation_aggregator.py [hunk 2]
```diff
@@ -26,64 +27,99 @@ def forward(self, x):
...
+
+    updater = chainer.training.StandardUpdater(train_iter, optimizer,
+                                               device=device)
 
     trainer = chainer.training.Trainer(updater, (1, 'epoch'))
 
     @extension.make_extension(
         trigger=(1, 'iteration'), priority=extension.PRIORITY_WRITER)
-    def rank_reporter(trainer):
+    def rank_reporter(trainer_):
```
Comment (shu65) [inline tests/chainermn_tests/extensions_tests/test_observation_aggregator.py]: Why did you change `trainer` to `trainer_`?

File: tests/chainermn_tests/extensions_tests/test_observation_aggregator.py [hunk 3]
```diff
@@ -26,64 +27,99 @@ def forward(self, x):
...
-        chainer.optimizers.Adam(), comm)
-    optimizer.setup(model)
 
-    train = xp.random.rand(10, 1).astype(np.float32)
+    if use_gpu:
+        # Use CuPy's Device class to force call cudaSetDevice()
+        chainer.cuda.get_device_from_id(comm.intra_rank).use()
+
+    device = get_device(comm.intra_rank if use_gpu else None, xp == chainerx)
+    device.use()
```
Comment (belldandyxtq) [inline tests/chainermn_tests/extensions_tests/test_observation_aggregator.py]: Unnecessary

File: tests/chainermn_tests/extensions_tests/test_observation_aggregator.py (hunk 2 above)
Comment (belldandyxtq) [reply tests/chainermn_tests/extensions_tests/test_observation_aggregator.py]: fix closure's parameter name that hid outer variables?
9296d5e14970699aa29619f838918f45c676ffe9

Comment (keisukefukuda) [reply tests/chainermn_tests/extensions_tests/test_observation_aggregator.py]: Yes, @belldandyxtq is right. if it is `trainer`, it hides the variable in the outer scope.

File: tests/chainermn_tests/extensions_tests/test_observation_aggregator.py (hunk 3 above)
Comment (keisukefukuda) [reply tests/chainermn_tests/extensions_tests/test_observation_aggregator.py]: fixed. thanks.

**Comment by keisukefukuda** [comment]
pfnci, test this please.

File: chainermn/extensions/observation_aggregator.py [hunk 4]
```diff
@@ -47,6 +47,10 @@ def compute_summary(self, trainer):
                 # use to native device as ChainerX array cannot
                 # be converted to numpy directly, which is what `to_cpu()` does
                 value.to_device("native")
+            elif isinstance(value, chx.ndarray) and \
+                    not value.device.name.startswith('native'):
+                raise ValueError(
+                    "observation aggregator does not support ChainerX ndarray")
```
[the remark this answers is not in the document]
Comment (keisukefukuda) [reply chainermn/extensions/observation_aggregator.py]: Fixed.

**Comment by keisukefukuda** [comment]
pfnci, test this  please.
